containers/go-postgres/.devcontainer/Dockerfile

```diff
@@ -10,10 +10,14 @@ RUN if [ "${NODE_VERSION}" != "none" ]; then su vscode -c "umask 0002 && . /usr/
 # RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
 #     && apt-get -y install --no-install-recommends <your-package-list-here>
 
-# [Optional] Add more packages to the below lines to use go get to install anything else you need
-USER vscode
-RUN go get -x github.com/lib/pq
-USER root
+# Copy go.mod and go.sum (if found) to a temp location so we update the environment. Also
+# copy "noop.txt" so the COPY instruction does not fail if no go.mod or go.sum exists.
+COPY go.mod* go.sum* .devcontainer/noop.txt /tmp/go-tmp/
+RUN if [ -f "/tmp/go-tmp/go.mod" ] && [ -f "/tmp/go-tmp/go.sum" ]; then \
+    umask 0002 && \
+    cd /tmp/go-tmp && \
+    go mod download; fi \
```

Comment:
Out of curiosity. Is it possible to download the required package as part of the test run instead of doing it here?

Reply:
good question @juzuluag -- actually, what this PR does (for CI) only uses the _test-project_ dirs instances of _go.mod_ & _go.sum_ during the test run (_test.sh_ & _F5-Debug_), whereas this `DockerFile` addition is more 'boilerplate' for how to have your "go modules auto-included for the whole container", in the spirit of what we've seen, and replicated, from [python-3-anaconda](https://github.com/microsoft/vscode-dev-containers/blob/aead89da029f1e2e6d9a353fd0f77bdf78451241/containers/python-3-anaconda/.devcontainer/Dockerfile#L7-L11) support for a workspace _environment.yml_, for example  

Meaning, that new section of the DockerFile is 'not active' until someone puts a _go.mod_ and _go.sum_ file in their workspace root -- this actually leads to the question we want answered from Chuck, about having that section _uncommented_ or _commented_